apps/aehttp/src/aehttpc_aeternity.erl

```diff
@@ -94,20 +103,79 @@ get_generation_by_height(NodeSpec, Height) ->
     get_request(Path, NodeSpec, 5000).
 
 
-%% TODO This function copied from aec_test_utils as that module is not available
-%% in normal builds
-%% TODO - wallet interaction of some kind to get privKey
-%% This is a horrible hack for now
-%% Maybe aec_preset_keys??
-%% {ok, Sig} = aec_preset_keys:sign_tx(SpendTx, AccountId)
-%% TODO: redo this whole thing
--define(VALID_PRIVK(K), byte_size(K) =:= 64).
-
-% sign_tx(Tx, NetworkId, Signer, SignModule) when is_binary(Signer) ->
-%     Bin0 = aetx:serialize_to_binary(Tx),
-%     BinForNetwork = aec_governance:add_custom_network_id(NetworkId, Bin0),
-%     {ok, Signature} = SignModule:sign_binary(BinForNetwork, Signer),
-%     aetx_sign:new(Tx, [Signature]).
+%%%=============================================================================
+%%% Pinning
+%%%=============================================================================
+
+-spec encode_parent_pin_payload(#{epoch => integer(), height => integer(), block_hash => binary()}) -> binary().
+encode_parent_pin_payload(#{epoch := Epoch, height := Height, block_hash := BlockHash}) ->
+    EpochHex = list_to_binary(erlang:integer_to_list(Epoch, 16)),
+    HeightHex = list_to_binary(erlang:integer_to_list(Height, 16)),
+    EncBlockHash = aeser_api_encoder:encode(key_block_hash, BlockHash),
+    <<EpochHex/binary, ":", HeightHex/binary, " ", EncBlockHash/binary>>.
+
+-spec decode_parent_pin_payload(binary()) -> #{epoch => integer(), height => integer(), block_hash => binary()}.
+decode_parent_pin_payload(Binary) ->
+    try
+        [HexEpoch, HexHeight, EncBlockHash] = binary:split(Binary, [<<":">>, <<" ">>], [global]),
+        Epoch = erlang:list_to_integer(binary_to_list(HexEpoch), 16),
+        Height = erlang:list_to_integer(binary_to_list(HexHeight), 16),
+        {ok, BlockHash} = aeser_api_encoder:safe_decode(key_block_hash, EncBlockHash),
+        #{epoch => Epoch, height => Height, block_hash => BlockHash}
+    catch 
+        _ -> {error, {bad_parent_pin_payload, Binary}}
+    end.
+
+encode_child_pin_payload(TxHash) ->
+    <<$p,$i,$n, TxHash/binary>>.
```

Comment:
```suggestion
    <<"pin", TxHash/binary>>.
```

Reply:
Fixed. In the commit queue.

Reply:
Possibly you need `"pin"/utf8` but otherwise it should work...